### PR TITLE
Support local properties in external tests

### LINKED ISSUE #3428: external test derby_tests failed when download junit
external test derby_tests failed when download junit
https://ci.adoptopenjdk.net/view/work-in-progress/job/grinder_sandbox_new/480/console

```
09:59:40  install_junit:
09:59:40        [get] Getting: http://repo1.maven.org/maven2/junit/junit/3.8.2/junit-3.8.2.jar
09:59:40        [get] To: /derby/tools/java/junit.jar
09:59:40        [get] Error opening connection java.io.IOException: Server returned HTTP response code: 501 for URL: http://repo1.maven.org/maven2/junit/junit/3.8.2/junit-3.8.2.jar
09:59:40        [get] Error opening connection java.io.IOException: Server returned HTTP response code: 501 for URL: http://repo1.maven.org/maven2/junit/junit/3.8.2/junit-3.8.2.jar
09:59:40        [get] Error opening connection java.io.IOException: Server returned HTTP response code: 501 for URL: http://repo1.maven.org/maven2/junit/junit/3.8.2/junit-3.8.2.jar
09:59:40        [get] Can't get http://repo1.maven.org/maven2/junit/junit/3.8.2/junit-3.8.2.jar to /derby/tools/java/junit.jar
```

### PR DESCRIPTION
**Pull request by sophia-guo**
Support local properties in external tests

Fix #3428 

Signed-off-by: Sophia Guo <sophia.gwf@gmail.com>

### CODE REVIEW

**Comment by sophia-guo** [comment]
Grinder:
 https://ci.adoptopenjdk.net/view/work-in-progress/job/grinder_sandbox_new/494/
https://ci.adoptopenjdk.net/view/work-in-progress/job/grinder_sandbox_new/489/ ( cancelled after the PR is verified)

File: external/dockerfile_functions.sh [hunk 1]
```diff
@@ -426,7 +427,10 @@ print_test_script() {
                 "\nCOPY ${test}/dockerfile/${script} /${script}" \
                 "\nCOPY test_base_functions.sh test_base_functions.sh\n" >> ${file}
     fi
-    
+    if [[ ! -z ${localPropertyFile} ]]; then
+        echo -e "# This is local perperty file needed to set up user preferred properties." \
```
Comment (smlambert) [inline external/dockerfile_functions.sh:431]:
```suggestion
        echo -e "# This local property file is needed to set up user preferred properties." \
```

**Review by smlambert — APPROVED** [review]
Aside from a suggestion to fix a minor spelling in a comment, LGTM, thanks @sophia-guo !

File: external/dockerfile_functions.sh (hunk 1 above)
Comment (sophia-guo) [reply external/dockerfile_functions.sh]: Thanks Shelley! Updated.

**Review by llxia — APPROVED** [review]
LGTM

**Pull request merged**